### ISSUE TITLE
Nyxt buffer's after crash lose owner 

### ISSUE THREAD

**Issue by mariari** [atlas-engineer/nyxt#1560]
**Describe the bug**

After Nyxt crashes  and you restart it. You may get a message like

```
<WARN> [13:13:02] Warning: Failed to restore history from /home/katya/.local/share/nyxt/history/default.lisp: Owner with identifier "11" does not exist
```

When you try to change buffers to any previous one you get something akin to

```
<WARN> [13:13:45] Warning: Error on separate thread: Owner with identifier "16" does not exist
```

The only solution is to kill your history. Any attempt to delete the buffer with `delete-buffers` Will result in this

```lisp
Unhandled SIMPLE-ERROR in thread #<SB-THREAD:THREAD "Prompter follow selection thread" RUNNING
                                    {1008721133}>:
  Owner with identifier "16" does not exist

Backtrace for: #<SB-THREAD:THREAD "Prompter follow selection thread" RUNNING {1008721133}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {10087B2743}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK SB-EXT:*INVOKE-DEBUGGER-HOOK* #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {10087B2743}>)
2: (INVOKE-DEBUGGER #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {10087B2743}>)
3: (ERROR "Owner with identifier ~s does not exist" "16")
4: (HISTORY-TREE:SET-CURRENT-OWNER #<HISTORY-TREE:HISTORY-TREE {10065EDF53}> "16")
5: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN WINDOW-SET-BUFFER))
6: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
7: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<FUNCTION (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN WINDOW-SET-BUFFER) {7F2A5C5A670B}> #<SB-THREAD:MUTEX "Anonymous recursive lock" owner: #<SB-THREAD:THREAD "Prompter follow selection thread" RUNNING {1008721133}>> T NIL)
8: (WINDOW-SET-BUFFER #<USER-WINDOW {1006D94943}> #<USER-WEB-BUFFER {1007688513}> :FOCUS NIL)
9: (SET-CURRENT-BUFFER #<USER-WEB-BUFFER {1007688513}> :FOCUS NIL)
10: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
11: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
12: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::RUN))
13: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
14: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::RUN))
15: (SB-THREAD::RUN)
16: ("foreign function: call_into_lisp")
17: ("foreign function: funcall1")

unhandled condition in --disable-debugger mode, quitting
<WARN> [13:14:49] Warning: Error: Joining thread failed: thread #<THREAD "Nyxt renderer thread" ABORTED {1007C18573}> did not return normally.


```

**Precise recipe to reproduce the issue**

Find a way to crash nyxt and corrupt the buffer history. It seems one possible crash that triggers it has a message such as

```lisp
CORRUPTION WARNING in SBCL pid 898337 tid 898343:
Memory fault at (nil) (pc=0x7fdf45fa5320, fp=0x7fde3400a250, sp=0x7fdf3e235cb0) tid 898343
The integrity of this image is possibly compromised.
Continuing with fingers crossed.
<WARN> [13:12:54] Warning: Error on GTK thread: Unhandled memory fault at #x0.
; 
; compilation unit aborted
;   caught 1 fatal ERROR condition

```

**Information**
- OS name+version:
  - Arch Linux 
- Graphics card and driver:
- Desktop environment / Window manager name+version:
  - Xmonad 
- How you installed Nyxt (Guix pack, package manager, build from source):
  - Master 
- Information from command copy-system-information:

  If you can't run copy-system-information, try `nyxt --system-information` from
  a shell.  It this still does not work, please provide the following:
  - Nyxt version (from =M-x nyxt-version= or =nyxt --version=):
  - Lisp implementation/version (if built from source):
  - Kernel name+version:
  - WebKitGTK+ | QtWebEngine version:

**Output when started from a shell**
```lisp
<INFO> [13:13:00] Listening to socket "/run/user/1000/nyxt/nyxt.socket".
Nyxt version 2.1.0-7-gbbdba4aa
<INFO> [13:13:00] Loading Lisp file "/home/katya/.config/nyxt/auto-config.lisp".
<INFO> [13:13:00] Loading Lisp file "/home/katya/.config/nyxt/init.lisp".
To load "slynk":
  Load 1 ASDF system:
    slynk
; Loading "slynk"

SLYNK's ASDF loader finished.<INFO> [13:13:01] Loading history of 233 URLs from "/home/katya/.local/share/nyxt/history/default.lisp".
<INFO> [13:13:02] Loading history of 233 URLs from "/home/katya/.local/share/nyxt/history/default.lisp".
<WARN> [13:13:02] Warning: Failed to restore history from /home/katya/.local/share/nyxt/history/default.lisp: Owner with identifier "11" does not exist
<WARN> [13:13:16] Warning: Error on separate thread: Owner with identifier "16" does not exist
<INFO> [13:13:39] Loading 43 bookmarks from "/home/katya/.local/share/nyxt/bookmarks.lisp".
<INFO> [13:13:42] Loading "https://duckduckgo.com/?q=doctor%20zilach".
<INFO> [13:13:42] Updating hostlist "/home/katya/.local/share/nyxt/hostlist-stevenblack.txt" from "https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts".
<INFO> [13:13:44] Finished loading "https://duckduckgo.com/?q=doctor+zilach&ia=web".
<WARN> [13:13:45] Warning: Error on separate thread: Owner with identifier "16" does not exist
<INFO> [13:13:46] Loading "https://www.freebsd.org/cgi/man.cgi?query=startx&sektion=1&manpath=freebsd-release-ports".
<INFO> [13:13:47] Finished loading "https://www.freebsd.org/cgi/man.cgi?query=startx&sektion=1&manpath=freebsd-release-ports".
<WARN> [13:13:48] Warning: Error on separate thread: Owner with identifier "16" does not exist
<WARN> [13:13:49] Warning: Error on separate thread: Owner with identifier "16" does not exist
<INFO> [13:13:57] Loading "https://duckduckgo.com/?q=doctor+zilach&ia=web".
<INFO> [13:13:57] Finished loading "https://duckduckgo.com/?q=%21g+doctor+zilch+underail".
<INFO> [13:13:57] Loading "https://duckduckgo.com/?q=%21g+doctor+zilch+underail".
<INFO> [13:13:57] Failed to load URL https://google.com/search?hl=en&q=doctor%20zilch%20underail in buffer 17.
<INFO> [13:13:57] Finished loading "https://duckduckgo.com/?q=%21g+doctor+zilch+underail".
<INFO> [13:13:57] Loading "https://duckduckgo.com/?q=%21g+doctor+zilch+underail".
<INFO> [13:13:58] Finished loading "https://www.google.com/search?hl=en&q=doctor%20zilch%20underail".
<INFO> [13:14:08] Loading "https://www.google.com/search?hl=en&q=doctor%20zilch%20underail".
<INFO> [13:14:09] Finished loading "https://steamcommunity.com/app/250520/discussions/0/1621724915803900311/".
<INFO> [13:14:16] Loading "https://steamcommunity.com/app/250520/discussions/0/1621724915803900311/".
<INFO> [13:14:16] Finished loading "https://www.google.com/search?hl=en&q=doctor%20zilch%20underail".
<INFO> [13:14:17] Loading "https://www.google.com/search?hl=en&q=doctor%20zilch%20underail".
<INFO> [13:14:19] Finished loading "https://steamcommunity.com/app/250520/discussions/0/1621724915803900311/?l=latam".
<INFO> [13:14:20] Loading "https://steamcommunity.com/app/250520/discussions/0/1621724915803900311/?l=latam".
<INFO> [13:14:20] Failed to load URL https://www.google.com/search?hl=en&q=doctor%20zilch%20underail in buffer 17.
<INFO> [13:14:20] Finished loading "https://www.google.com/search?hl=en&q=doctor%20zilch%20underail".
<INFO> [13:14:22] Loading "https://www.google.com/search?hl=en&q=doctor%20zilch%20underail".
<INFO> [13:14:23] Finished loading "https://www.google.com/search?hl=en&q=doctor%20zilch%20underail".
<WARN> [13:14:23] Warning: Error on separate thread: Owner with identifier "16" does not exist
<WARN> [13:14:24] Warning: Error on separate thread: Owner with identifier "16" does not exist
<WARN> [13:14:24] Warning: Error on separate thread: Owner with identifier "16" does not exist
<WARN> [13:14:24] Warning: Error on separate thread: Owner with identifier "16" does not exist
<WARN> [13:14:25] Warning: Error on separate thread: Owner with identifier "16" does not exist
]]<WARN> [13:14:36] Warning: Error on separate thread: Owner with identifier "3" does not exist
<WARN> [13:14:36] Warning: Error on separate thread: Owner with identifier "3" does not exist
<WARN> [13:14:36] Warning: Error on separate thread: Owner with identifier "3" does not exist
<WARN> [13:14:37] Warning: Error on separate thread: Owner with identifier "3" does not exist
<WARN> [13:14:37] Warning: Error on separate thread: Owner with identifier "3" does not exist
Unhandled SIMPLE-ERROR in thread #<SB-THREAD:THREAD "Prompter follow selection thread" RUNNING
                                    {1008721133}>:
  Owner with identifier "16" does not exist

Backtrace for: #<SB-THREAD:THREAD "Prompter follow selection thread" RUNNING {1008721133}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {10087B2743}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK SB-EXT:*INVOKE-DEBUGGER-HOOK* #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {10087B2743}>)
2: (INVOKE-DEBUGGER #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {10087B2743}>)
3: (ERROR "Owner with identifier ~s does not exist" "16")
4: (HISTORY-TREE:SET-CURRENT-OWNER #<HISTORY-TREE:HISTORY-TREE {10065EDF53}> "16")
5: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN WINDOW-SET-BUFFER))
6: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
7: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<FUNCTION (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN WINDOW-SET-BUFFER) {7F2A5C5A670B}> #<SB-THREAD:MUTEX "Anonymous recursive lock" owner: #<SB-THREAD:THREAD "Prompter follow selection thread" RUNNING {1008721133}>> T NIL)
8: (WINDOW-SET-BUFFER #<USER-WINDOW {1006D94943}> #<USER-WEB-BUFFER {1007688513}> :FOCUS NIL)
9: (SET-CURRENT-BUFFER #<USER-WEB-BUFFER {1007688513}> :FOCUS NIL)
10: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
11: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
12: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::RUN))
13: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
14: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::RUN))
15: (SB-THREAD::RUN)
16: ("foreign function: call_into_lisp")
17: ("foreign function: funcall1")

unhandled condition in --disable-debugger mode, quitting
<WARN> [13:14:49] Warning: Error: Joining thread failed: thread #<THREAD "Nyxt renderer thread" ABORTED {1007C18573}> did not return normally.


```

**Comment by Ambrevar** [comment]
Since you are running 2.1, the history should not longer be corrupted on crash.  

Something else is wrong here, and I've experience it as well.  I'll work on it, this is top priority.

**Comment by Ambrevar** [comment]
Can you test #1560 and report if it fixes it for you?

**Comment by Ambrevar** [comment]
The symptom of this issue is the double line:

```
<INFO> [13:13:02] Loading history of 233 URLs from "/home/katya/.local/share/nyxt/history/default.lisp".
```

The history was restored twice, thus corrupting everything (because new buffer IDs are always different) :(

**Comment by Ambrevar** [comment]
Closing as this should be fixed by #1560.
